components/datetime/src/time_zone.rs

```diff
@@ -89,27 +88,27 @@ where
 ///
 /// // let value = tzf.format_to_string(&time_zone);
 /// // ```
-// TODO(#622) Make TimeZoneFormat public once we have a clean way to provide it options.
-pub(super) struct TimeZoneFormat<'data> {
+pub struct TimeZoneFormat<'data> {
     /// The pattern to format.
-    pub(super) patterns: DataPayload<'data, PatternPluralsFromPatternsV1Marker>,
+    pub(super) patterns: Option<DataPayload<'data, PatternPluralsFromPatternsV1Marker>>,
     /// The data that contains meta information about how to display content.
-    pub(super) zone_formats: DataPayload<'data, provider::time_zones::TimeZoneFormatsV1Marker>,
+    pub zone_formats: DataPayload<'data, provider::time_zones::TimeZoneFormatsV1Marker>,
     /// The exemplar cities for time zones.
-    pub(super) exemplar_cities:
-        Option<DataPayload<'data, provider::time_zones::ExemplarCitiesV1Marker>>,
+    pub exemplar_cities: Option<DataPayload<'data, provider::time_zones::ExemplarCitiesV1Marker>>,
     /// The generic long metazone names, e.g. Pacific Time
-    pub(super) mz_generic_long:
+    pub mz_generic_long:
         Option<DataPayload<'data, provider::time_zones::MetaZoneGenericNamesLongV1Marker>>,
     /// The generic short metazone names, e.g. PT
-    pub(super) mz_generic_short:
+    pub mz_generic_short:
         Option<DataPayload<'data, provider::time_zones::MetaZoneGenericNamesShortV1Marker>>,
     /// The specific long metazone names, e.g. Pacific Daylight Time
-    pub(super) mz_specific_long:
+    pub mz_specific_long:
         Option<DataPayload<'data, provider::time_zones::MetaZoneSpecificNamesLongV1Marker>>,
     /// The specific short metazone names, e.g. Pacific Daylight Time
-    pub(super) mz_specific_short:
+    pub mz_specific_short:
         Option<DataPayload<'data, provider::time_zones::MetaZoneSpecificNamesShortV1Marker>>,
+    /// The config of TimeZoneFormat
+    pub config: Option<TimeZoneFormatConfig>,
```

Comment:
> issue

Sorry I wasn't clear about this. Only the public API (function calls) should become fully `pub`. 

We don't want these data members to be fully exposed. They can go back to being `pub(super)`.